src/parser/core/modules/CastTime.ts

```diff
@@ -31,52 +32,128 @@ export default class CastTime extends Analyser {
 	}
 
 	private onApplySwiftcast(): void {
-		this.scIndex = this.set('all', 0)
+		this.scIndex = this.setInstantCastAdjustment()
 	}
 
 	private onRemoveSwiftcast(): void {
 		this.reset(this.scIndex)
 		this.scIndex = null
 	}
 
-	/** @deprecated */
-	public setFflogs(actions: number[] | 'all', castTime: number, start: number = this.parser.currentTimestamp, end: number | null = null): number {
-		return this.set(actions, castTime, this.parser.fflogsToEpoch(start), end ? this.parser.fflogsToEpoch(end) : null)
+	/**
+	 * setTimeAdjustment Module-compatibility function
+	 * @deprecated
+	 */
+	public setTimeAdjustmentFflogs(actions: number[] | 'all', adjustment: number, start: number = this.parser.currentTimestamp, end: number | null = null): number {
+		return this.setTimeAdjustment(actions, adjustment, this.parser.fflogsToEpoch(start), end ? this.parser.fflogsToEpoch(end) : null)
 	}
-	public set(actions: number[] | 'all', castTime: number, start: number = this.parser.currentEpochTimestamp, end: number | null = null): number {
+	/**
+	 * Sets a cast time adjustment for a flat time amount per cast (See: Lightspeed, Dreadwyrm Trance, etc.)
+	 * @param actions The actions this adjustment applies to. Either an array of IDs, or the string 'all'
+	 * @param adjustment The amount of time that cast times are adjustmented by
+	 * @param start The beginning of the adjustment time range. Defaults to the current epoch timestamp
+	 * @param end The end of the adjustment time range. May be left null if the end of the range is not yet known
+	 * @returns The index number within the cast time adjustments collection, can be used to reset/end this adjustment later
+	 */
+	public setTimeAdjustment(actions: number[] | 'all', adjustment: number, start: number = this.parser.currentEpochTimestamp, end: number | null = null): number {
+		return this.set(actions, 'time', adjustment, start, end)
+	}
+
+	/**
+	 * setInstantCastAdjustment Module-compatibility function
+	 * @deprecated
+	 */
+	public setInstantCastAdjustmentFflogs(actions: number[] | 'all' = 'all', start: number = this.parser.currentTimestamp, end: number | null = null): number {
+		return this.set(actions, 'percentage', 0, this.parser.fflogsToEpoch(start), end ? this.parser.fflogsToEpoch(end) : null)
+	}
+	/**
+	 * Shorthand function for setting all casts to instant (ie. Swiftcast, Triplecast)
+	 * @actions The actions this adjustment applies to. Either an array of IDs, or the string 'all'. Defaults to 'all'
+	 * @param start
```

Comment:
```suggestion
	 * @param start The beginning of the adjustment time range. Defaults to the current epoch timestamp
	 * @param end The end of the adjustment time range. May be left null if the end of the range is not yet known
	 * @returns The index number within the cast time adjustments collection, can be used to reset/end this adjustment later
```

GitHub isn't letting me do a multi-line comment right, but I think you had some line drift on this doc comment

Reply:
Copy paste, thou hast failed me